proxy/src/main/java/com/wavefront/agent/listeners/FeatureCheckUtils.java

```diff
@@ -71,13 +71,50 @@ public static boolean isFeatureDisabled(Supplier<Boolean> featureDisabledFlag, S
                                           @Nullable Counter discardedCounter,
                                           @Nullable StringBuilder output,
                                           @Nullable FullHttpRequest request) {
+    return isFeatureDisabled(featureDisabledFlag, message, discardedCounter, 1, output, request);
+  }
+
+  /**
+   * Check whether feature disabled flag is set, log a warning message, increment the counter by
+   * increment.
+   *
+   * @param featureDisabledFlag Supplier for feature disabled flag.
+   * @param message             Warning message to log if feature is disabled.
+   * @param discardedCounter    Counter for discarded items.
+   * @param increment           The amount by which the counter will be increased.
+   * @return true if feature is disabled
+   */
+  public static boolean isFeatureDisabled(Supplier<Boolean> featureDisabledFlag,
+                                          String message,
+                                          Counter discardedCounter,
+                                          long increment) {
```

Comment:
Why we need this ?

Reply:
This was added so I could increment `discardedCounter` by the total number of discarded spans in the two Listeners:
- [OtlpGrpcTraceHandler](https://github.com/oppegard/wavefront-proxy/blob/0b10d6904c77064792894cd2ec2f38991fa93a52/proxy/src/main/java/com/wavefront/agent/listeners/otlp/OtlpGrpcTraceHandler.java#L126)
- [OtlpHttpHandler](https://github.com/oppegard/wavefront-proxy/blob/0b10d6904c77064792894cd2ec2f38991fa93a52/proxy/src/main/java/com/wavefront/agent/listeners/otlp/OtlpHttpHandler.java#L122)

Reply:
No need, this is to reduce the log messages, and this change will increment it.

Reply:
I'm not sure how you're suggesting to accurately increment the discarded spans counter. Should I use the existing `isFeatureDisabled()` method that increments the counter by one, and then manually increment the discarded spans counter by `spansCount-1` back in the in listener?

With the new method, I can check if the span feature is disabled and increment the total number of spans that will be discarded in the batch in one go: https://github.com/oppegard/wavefront-proxy/blob/0b10d6904c77064792894cd2ec2f38991fa93a52/proxy/src/main/java/com/wavefront/agent/listeners/otlp/OtlpGrpcTraceHandler.java#L123-L126.